proto/src/dns_handle.rs

```diff
@@ -268,14 +267,15 @@ where
                     let query_id = query_id.expect("query_id should have been set above");
                     message.set_id(query_id);
 
+                    let now = SystemTime::now().duration_since(UNIX_EPOCH)
+                        .map_err(|_| "Current time is before the Unix epoch.".into())?
+                        .as_secs();
+                    let now = now as u32; // XXX: truncates u64 to u32.
```

Comment:
When `TryFrom` stabilizes we could change this to a `u32::try_from(now).or_else(u32::max)`